### PR TITLE
Campaign part 2

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Actually testable edition

Continuing to build on the last pr, this should be set for an initial test to make sure everything works, although missions are still bare due to the demand for more maps.

Rundown to help people with currently tested missions:
**Campaign Status screen:**
This is where all the campaign specific stuff is viewable, and can be accessed in the IC tab.

The Faction Leader can activate everything, while everyone else has read only access (SL's can now activate certain assets themselves, such as equipment and mortars).

**Missions:**
Campaign is divided into missions. Every mission has its own objectives and z-level. Winning a mission provided various benefits, and allows the winner to select the next mission.

Available missions can be seen in the Campaign Status panel.

**Attrition Points:**
AP = the currency of the game. 1 AP = 1 respawn. The faction leader chooses how many respawns to allocate to a mission BEFORE IT BEGINS. Any attrition unused at the end of the mission is lost.

AP can also be spent on Assets, to give a variety of buffs.

**Assets:**
Assets are various beneficial (or in some case detrimental) bonuses for your faction. Some are quite simple like equipment (press button for b18, etc) while others give more strategic benefits, such as allied troops, attrition gain modifiers, etc.

All members of a faction can see all assets owned or available for purchase in the campaign status screen.

**Firesupport:**
Squad leaders and the FC can call down firesupport via their tac binos, if the assets have been activated.
Simply choose a firemode, lase, and watching the fireworks.

Firesupport is shared across the faction, so you are sharing ammo count and cooldowns.

Fire support assets can be activated multiple times per mission, but **unused uses** are lost at the end of the mission.
**Mission types:**
Note these are just what are being currently tested, there are many more to come in the full release.

**Combat Patrol:**
TDM rules, wipe the enemy for a major victory, end the game with more kills for a minor.
NEW WIN CONDITIONS:
TDM games now have sensor towers that can be captured for points. Hit the point limit for an immediate major victory.
Cap towers for points, or to uncap enemy towers. Capture progress can be cancelled by the enemy team as well.

**Mech wars:**
Literally combat patrol, but with mechs and anti mech RPG's

**Fire support destroy missions:**
One team is trying to blow up the fire support of the other team (howitzers or MLRS's currently). Defending team deploys first for a little bit of time to setup defences. Attacking team need to blow them all up with C4.

**Capture mission:**
Lolly scramble for valuable phoron crates. Click a crate to activate it, after 30 seconds it will be fultoned to safety. You can cancel this process for the enemy team if you can get to the crate.
At the end of the mission, teams are awarded bonus attrition for each crate captured.

FAQ:
How do I respawn?
Respawns in campaign are always by the respawn ACTION BUTTON not the verb. Respawns can be done at any time between missions, but are only available mid missions if you have attrition available. Attrition can be seen in the status tab.

How do I deploy?
Deploy by clicking the red deploy circles on the tadpole(s) (or shift clicking as mech).
Alternatively, TGMC can deploy via drop pods (including mech pods) if their leader has activated the ability to do so for that mission. SOM instead have a large AOE teleporter.

How do I rearm/get more ammo?
Currently its intended that you can't get a freshload out between missions, so scavaging is recommended. Additionally, your leaders can activate weapon or supply assets to help rearm.

## Changelog
:cl:
add: More progress to the Campaign gamemode
balance: You can no longer throw bump mechs or other high resist objects
balance: Drop pods are more resistant to projectile damage
imageadd: Updated tac bino sprites (previous pr)
imageadd: New sprites and animations for drop pods
fix: Fixed the minimap not working when you move inside another object
fix: Fixed some various minor bugs/runtimes
/:cl:

### CODE REVIEW

**Comment by Pariah919** [comment]
Can you put into the CL that campaign changed tac!bino sprites?

**Comment by Lumipharon** [comment]
giving campaign a test round for bug finding.

**Comment by dopamiin0** [comment]
consolidating a lot of feedback from myself (and some from other people who played). not putting bugs here

som cas just outclasses tgmc cas. incen minis and the SOM gunrun are both just generally better than the tgmc options. would be nice for tgmc cas to have a strong zoning option analogous to the incen minis

SLs should all be able to activate purchased assets. would be nice if they could also purchase them

cas/mortar options should have some means of checking remaining uses/cooldown

expensive one use ob asset when
attack rouny squad asset when (maybe as a mission reward)

activating assets should have a big announcement for the respective team and not just a small line of easily missable red text. ideally the message should also, like, explain some less obvious things - i don't know what the fuck 'teleporter activated' means as som. where do i go, what does it do

more announcements that explain things in general would be nice

keeping your gear when you respawn is... interesting? especially given it ALSO lets you pick a new kit. game mode in general feels somewhat snowbally, partially because of this - it seems like it might be hard for a team on the back foot to recover, when the winning team is accumulating both bonus assets from mission rewards and keeping powerful items. maybe some sort of catchup mechanic? bonus attrition points based on victory point disparity? dunno

hard to give more feedback cuz we only had part of one round and it was bugged but the mode was honestly super fun and promising and im really looking forward to more of it

**Comment by Pariah919** [comment]
Commander call ins should probably work like RO2/RS2 call ins, big telegraph for your team and an obvious sound for the enemy team as it rolls in, some spotting call ins would be nice.

Keeping gear is.. ehhh. I think it causes more issues than it does anything "Cool" wise.

Also, as much as I hate to say this- SOM is blatantly just better than marines, their guns/specialized+general gear are made for HvH while Marine guns... Aren't. Veterans give you more options than smartgunners really do. If you're gonna keep SoM around as the main antagonist you'll probably either need to bring them down a bit or make hvh versions of the Marine guns, personally speaking the asymmetry massively favors SoM currently as most marine snowflake gear was designed around req, while SoM snowflake gear was designed around HvH design.

Maps also need to be narrower and smaller to have more consistent engagements, but maps are hellish to make so is understandable to not have it work well.

Generally teams should win based off the fact they play better, not snowball like Nuclear War.

I would personally probably remove the TDM misson and focus entirely on breakthrough and capture zone modes since TDM just is ass in top down shooters, you want big spectacle engagements and maps that draw all people towards one general area, not have them run around like idiots trying to RDM each other in random parts of the map.

**Comment by FosterGrime-Heymynameslouie** [comment]
"honestly this would be really cool if it involved literally anything else other than hvh  (hvx)"
-fumo, in a 2 am sleep deprived discord call

"wouldnt it make more sense if som was the one using aerial rappel and tg had the chronosphere? they already have teleportation tech and all"
-me, in the same 2 am sleep deprived discord call

**Comment by TheUSSWatercooler** [comment]
Powernet on Patrick's Rest is a bit busted.
SMESs will flicker between sending and not sending after repairing generators, and the dam bridge's APC (engines) isn't wired to powernet at all.
No APC actually receives power, but says there is power in the net.

**Comment by imsxz** [comment]
the normal modes are really fun IMO, the mech mode is kinda aids

**Comment by chesse20** [comment]
I don't think the playerbase plays Marine vs Marine gamemodes enough to be good at them,
i think we need weighted voting so that if Campaign or another gamemode is 15% of the vote then it gets a chance to be picked randomly

**Comment by Mannybrado** [comment]
> I don't think the playerbase plays Marine vs Marine gamemodes enough to be good at them, i think we need weighted voting so that if Campaign or another gamemode is 15% of the vote then it gets a chance to be picked randomly instead of the winning gamemode always being picked otherwise everyone will keep playing nuclear war.

people do this because nuclear war is better and more fun.
until hvh is fun, this will always be the case. forcing people to play what they do not want to play will simply kill the fucking server.

**Comment by Lumipharon** [comment]
good catch

**Comment by dopamiin0** [comment]
Missions should have their max time halved, especially for missions where the win condition for one team requires time to run out. 20 minutes is unbelievably long.

If you're not gonna let people re-pick loadouts, at least add an ammo vendor. As-is, if you didn't loot anything particularly valuable the best strat is to just kill yourself so you can get new shit.

**Comment by FosterGrime-Heymynameslouie** [comment]
a variant of campaign focused on xenos would probably also be realaly cool

**Comment by Oscari-pog** [comment]
Some sort of a catch-up mechanic for the losing team is definitely necessary.

File: code/game/objects/effects/landmarks/landmarks.dm [hunk 1]
```diff
@@ -466,18 +466,101 @@
...
+	///Faction this belongs to for minimap purposes
+	var/faction = FACTION_TERRAGOV
+
+	var/minimap_icon = "patrol_1"
 
 /obj/effect/landmark/patrol_point/Initialize(mapload)
 	. = ..()
 	//adds the exit points to the glob, and the start points link to them in lateinit
 	GLOB.patrol_point_list += src
+	if(!(SSticker?.mode.flags_round_type & MODE_TWO_HUMAN_FACTIONS))
```
Comment (Lumipharon) [inline code/game/objects/effects/landmarks/landmarks.dm:480]:
```suggestion
	if(!(SSticker?.mode?.flags_round_type & MODE_TWO_HUMAN_FACTIONS))
```

**Comment by Fr3nchFr13s** [comment]
Let players choose whether to fight SoM or ICC.

SoM is nice, don't get me wrong, but can get stale after multiple back-to-back rounds.

File: code/game/objects/effects/landmarks/landmarks.dm [hunk 2]
```diff
@@ -466,18 +466,101 @@
 //Combat patrol spawn in spots
 /obj/effect/landmark/patrol_point
 	name = "Patrol exit point"
-	//ID to link with an associated start point
+	///ID to link with an associated start point
 	var/id = null
+	///Faction this belongs to for minimap purposes
+	var/faction = FACTION_TERRAGOV
+
+	var/minimap_icon = "patrol_1"
```
Comment (TiviPlus) [inline code/game/objects/effects/landmarks/landmarks.dm:474]: autodoc

File: code/game/objects/machinery/computer/droppod_control.dm [hunk 3]
```diff
@@ -32,8 +32,8 @@
 				var/obj/structure/droppod/pod = p
 				if(!length(pod.buckled_mobs))
 					continue
-				var/predroptime = rand(5, 1 SECONDS)	//Randomize it a bit so its staggered
-				addtimer(CALLBACK(pod, TYPE_PROC_REF(/obj/structure/droppod, launchpod), pod.buckled_mobs[1]), predroptime)
+				var/predroptime = rand(3, 1 SECONDS)	//Randomize it a bit so its staggered
+				addtimer(CALLBACK(pod, TYPE_PROC_REF(/obj/structure/droppod, start_launch_pod), pod.buckled_mobs[1]), predroptime)
```
Comment (TiviPlus) [inline code/game/objects/machinery/computer/droppod_control.dm:36]: ????

File: code/game/objects/structures/droppod.dm [hunk 4]
```diff
@@ -125,10 +127,9 @@ GLOBAL_LIST_INIT(blocked_droppod_tiles, typecacheof(list(/turf/open/space/transi
 	target_x = new_x
 	target_y = new_y
 	var/mob/notified_user = LAZYACCESS(buckled_mobs, 1)
-	if(notified_user)
-		. = checklanding(notified_user)
-		if(.)
-			balloon_alert(notified_user, "Coordinates updated")
+	. = checklanding(notified_user)
+	if(notified_user && .)
+		balloon_alert(notified_user, "Coordinates updated")
```
Comment (TiviPlus) [inline code/game/objects/structures/droppod.dm:132]: ???

File: code/game/objects/structures/droppod.dm [hunk 5]
```diff
@@ -209,7 +210,23 @@ GLOBAL_LIST_INIT(blocked_droppod_tiles, typecacheof(list(/turf/open/space/transi
...
+///actually launches the pod
+/obj/structure/droppod/proc/launch_pod(mob/user)
+	if(!launch_allowed)
+		drop_state = DROPPOD_READY
+		GLOB.droppod_list += src
+		for(var/atom/movable/deployed AS in contents)
+			deployed.forceMove(loc)
+		update_icon()
+		if(user)
+			to_chat(user, span_notice("Error. Ship calibration unavailable. Please %#&ç:*"))
```
Comment (TiviPlus) [inline code/game/objects/structures/droppod.dm:227]: why not ballon?

File: code/game/objects/structures/droppod.dm [hunk 6]
```diff
@@ -13,14 +10,14 @@ GLOBAL_LIST_INIT(blocked_droppod_tiles, typecacheof(list(/turf/open/space/transi
...
+	icon_state = "singlepod_green"
 	density = TRUE
 	anchored = TRUE
 	layer = ABOVE_OBJ_LAYER
 	resistance_flags = XENO_DAMAGEABLE
 	interaction_flags = INTERACT_OBJ_DEFAULT|INTERACT_POWERLOADER_PICKUP_ALLOWED_BYPASS_ANCHOR
-	soft_armor = list(MELEE = 50, BULLET = 70, LASER = 70, ENERGY = 100, BOMB = 70, BIO = 100, FIRE = 0, ACID = 0)
-	max_integrity = 50
+	soft_armor = list(MELEE = 25, BULLET = 80, LASER = 80, ENERGY = 90, BOMB = 70, BIO = 100, FIRE = 0, ACID = 0)
+	max_integrity = 75
```
Comment (TiviPlus) [inline code/game/objects/structures/droppod.dm:20]: ???

File: code/game/objects/structures/droppod.dm [hunk 7]
```diff
@@ -327,8 +344,8 @@ GLOBAL_LIST_INIT(blocked_droppod_tiles, typecacheof(list(/turf/open/space/transi
 			continue
 		for(var/mob/dropper AS in pod.buckled_mobs)
 			dropper.play_screen_text("<span class='maptext' style=font-size:24pt;text-align:center valign='top'><u>DROP UPDATED:</u></span><br>COMMENCING MASS DEPLOYMENT", /atom/movable/screen/text/screen_text/command_order)
-		var/predroptime = rand(4 SECONDS, 5 SECONDS) //Randomize it a bit so its staggered
-		addtimer(CALLBACK(pod, TYPE_PROC_REF(/obj/structure/droppod, launchpod), LAZYLEN(pod.buckled_mobs) ? pod.buckled_mobs[1] : null, TRUE), predroptime)
+		var/predroptime = rand(1.5 SECONDS, 2.5 SECONDS) //Randomize it a bit so its staggered
+		addtimer(CALLBACK(pod, TYPE_PROC_REF(/obj/structure/droppod, start_launch_pod), LAZYLEN(pod.buckled_mobs) ? pod.buckled_mobs[1] : null, TRUE), predroptime)
```
Comment (TiviPlus) [inline code/game/objects/structures/droppod.dm:348]: ???

File: code/game/turfs/walls/wall_types.dm [hunk 8]
```diff
@@ -17,6 +17,7 @@
 	///do we have bits of decoration to add to the walls?
 	var/decorated_wall = TRUE
 
+/* this completely breaks hull walls in HvH, will fix in a separate pr because wall overlays are horribly shitcode
 /turf/closed/wall/mainship/smooth_icon()
```
Comment (TiviPlus) [inline code/game/turfs/walls/wall_types.dm:21]: ????????

File: code/modules/mob/dead/observer/observer.dm [hunk 9]
```diff
@@ -237,6 +237,7 @@ GLOBAL_VAR_INIT(observer_default_invisibility, INVISIBILITY_OBSERVER)
 /mob/proc/ghostize(can_reenter_corpse = TRUE, aghosting = FALSE)
 	if(!key || isaghost(src))
 		return FALSE
+	SEND_SIGNAL(SSdcs, COMSIG_MOB_GHOSTIZE, src, can_reenter_corpse)
```
Comment (TiviPlus) [inline code/modules/mob/dead/observer/observer.dm:240]: why is this needed for set_death_time

File: code/modules/projectiles/sentries.dm [hunk 10]
```diff
@@ -15,7 +15,7 @@
 	var/knockdown_threshold = 150
 
 	///List of targets that can be shot at
-	var/list/mob/living/potential_targets = list()
+	var/list/potential_targets = list()
```
Comment (TiviPlus) [inline code/modules/projectiles/sentries.dm:18]: cast this as atom properly

File: code/modules/vehicles/mecha/_mecha.dm [hunk 11]
```diff
@@ -555,8 +555,8 @@
 ///Displays a special speech bubble when someone inside the mecha speaks
 /obj/vehicle/sealed/mecha/proc/display_speech_bubble(datum/source, list/speech_args)
 	SIGNAL_HANDLER
-	var/list/speech_bubble_recipients = get_hearers_in_view(7,src)
-	for(var/mob/M in speech_bubble_recipients)
+	var/list/speech_bubble_recipients = list()
+	for(var/mob/M in get_hearers_in_view(7,src))
 		if(M.client)
```
Comment (TiviPlus) [inline code/modules/vehicles/mecha/_mecha.dm:560]: ???

File: code/modules/vehicles/mecha/_mecha.dm [hunk 12]
```diff
@@ -478,12 +478,12 @@
 	if(LAZYACCESS(modifiers, MIDDLE_CLICK))
 		set_safety(user)
 		return COMSIG_MOB_CLICK_CANCELED
+	if(modifiers[SHIFT_CLICK]) //Allows things to be examined.
+		return target.mech_shift_click(src, user)
```
Comment (TiviPlus) [inline code/modules/vehicles/mecha/_mecha.dm:482]: Why?

File: code/modules/vehicles/mecha/mecha_mob_interaction.dm [hunk 13]
```diff
@@ -42,13 +42,21 @@
 		return FALSE
 	newoccupant.drop_all_held_items()
 	add_occupant(newoccupant)
-	newoccupant.forceMove(src)
+	if(newoccupant.loc != src)
+		newoccupant.forceMove(src)
 	newoccupant.update_mouse_pointer()
```
Comment (TiviPlus) [inline code/modules/vehicles/mecha/mecha_mob_interaction.dm:47]: ???

**Review by TiviPlus — CHANGES_REQUESTED** [review]
I have a refresh reminder as I type this so I assume theres more I missed

File: code/game/area/campaign_maps/som_base.dm [hunk 14]
```diff
@@ -0,0 +1,342 @@
+//rocinanteBase AREAS//
+/area/rocinante_Base
```
[the remark this answers is not in the document]
Comment (Lumipharon) [reply code/game/area/campaign_maps/som_base.dm:2]: done

File: code/controllers/subsystem/minimaps.dm [hunk 15]
```diff
@@ -299,9 +302,34 @@ SUBSYSTEM_DEF(minimaps)
...
+	RegisterSignal(loc, COMSIG_ATOM_EXITED, PROC_REF(cancel_override_minimap_tracking))
+
+///Stops minimap override tracking
+/atom/movable/proc/cancel_override_minimap_tracking(atom/movable/source, atom/movable/mover)
+	SIGNAL_HANDLER
+	if(mover != src)
+		return
+	var/image/blip = SSminimaps.images_by_source[src]
+	blip.UnregisterSignal(source, COMSIG_MOVABLE_MOVED)
+	UnregisterSignal(source, COMSIG_ATOM_EXITED)
```
[the remark this answers is not in the document]
Comment (Lumipharon) [reply code/controllers/subsystem/minimaps.dm:331]: Ivan thought it was fine, but happy to do something else that works better.

File: code/game/objects/effects/landmarks/landmarks.dm (hunk 2 above)
Comment (Lumipharon) [reply code/game/objects/effects/landmarks/landmarks.dm:474]: done

File: code/game/objects/machinery/cic_maptable.dm [hunk 16]
```diff
@@ -150,8 +151,8 @@
 	SIGNAL_HANDLER
 	if(!isnum(new_z))
 		return
+	QDEL_NULL(map)
```
[the remark this answers is not in the document]
Comment (Lumipharon) [reply code/game/objects/machinery/cic_maptable.dm:154]: to clear it out of every user that may be using it when the z-level changes. Although as I say this I realise this is probably a terrible way of doing it. Any thoughts on a clean way of doing this?

File: code/game/objects/machinery/computer/droppod_control.dm (hunk 3 above)
Comment (Lumipharon) [reply code/game/objects/machinery/computer/droppod_control.dm:36]: What? Pod drops where changed slightly due to the new animations, this is just aligning the console drop with the change.

File: code/game/objects/structures/droppod.dm (hunk 4 above)
Comment (Lumipharon) [reply code/game/objects/structures/droppod.dm:132]: Pods can get coords set without a mob occupant, and we need to know if the target coords were successfully updated or not regardless of that.

File: code/game/objects/structures/droppod.dm (hunk 5 above)
Comment (Lumipharon) [reply code/game/objects/structures/droppod.dm:227]: Existing alerts are chat, so I left it for consistancy. I can update them all if you want though.

File: code/game/objects/structures/droppod.dm (hunk 6 above)
Comment (Lumipharon) [reply code/game/objects/structures/droppod.dm:20]: tougher in hvh, same effective hp for melee.

File: code/game/objects/structures/droppod.dm (hunk 7 above)
Comment (Lumipharon) [reply code/game/objects/structures/droppod.dm:348]: pods have new animations

File: code/game/turfs/walls/wall_types.dm (hunk 8 above)
Comment (Lumipharon) [reply code/game/turfs/walls/wall_types.dm:21]: Yeah this code was scuffed, and was causing unplayable bugs for hvh so I couldn't test without disabling it. 

I can just remove the code entirely/revert the pr that caused it, because a proper fix is needed in another pr (wall icon code is super wack in general though).

File: code/modules/mob/dead/observer/observer.dm (hunk 9 above)
Comment (Lumipharon) [reply code/modules/mob/dead/observer/observer.dm:240]: Because people were just ghosting to bypass the timer.

File: code/modules/projectiles/sentries.dm (hunk 10 above)
Comment (Lumipharon) [reply code/modules/projectiles/sentries.dm:18]: done

File: code/modules/vehicles/mecha/_mecha.dm (hunk 11 above)
Comment (Lumipharon) [reply code/modules/vehicles/mecha/_mecha.dm:560]: This was causing runtimes everytime someone spoke in/near occupied mechs, as get_hearers_in_view was adding mechs or some other stuff to speech_bubble_recipients, instead of just the occupants.

File: code/modules/vehicles/mecha/_mecha.dm (hunk 12 above)
Comment (Lumipharon) [reply code/modules/vehicles/mecha/_mecha.dm:482]: so they can click on things without punching them.

File: code/modules/vehicles/mecha/mecha_mob_interaction.dm (hunk 13 above)
Comment (Lumipharon) [reply code/modules/vehicles/mecha/mecha_mob_interaction.dm:47]: This was causing repeat forcemove calls which caused issues with another change. I think minimap stuff.
